addon/webextension/background.js

```diff
@@ -1,129 +1,8 @@
-  *   correctly at Parquet at s.t.m.o.
-  *
-  *   Bold claim:  catching errors here
-  *
-  */
-function telemetry (data) {
-  function throwIfInvalid (obj) {
-    // Check: all keys and values must be strings,
-    for (const k in obj) {
-      if (typeof k !== 'string') throw new Error(`key ${k} not a string`);
-      if (typeof obj[k] !== 'string') throw new Error(`value ${k} ${obj[k]} not a string`);
-    }
-    return true
-  }
-  throwIfInvalid(data);
-  return msgStudyUtils("telemetry", data);
-}
-
-
-class BrowserActionButtonChoiceFeature {
-  /**
-    * - set image, text, click handler (telemetry)
-    * - tell Legacy Addon to send
-    */
-  constructor(variation) {
-    console.log("initilizing BrowserActionButtonChoiceFeature:", variation.name);
-    this.timesClickedInSession = 0;
-
-    // modify BrowserAction (button) ui for this particular {variation}
-    console.log("path:", `icons/${variation.name}.svg`)
-    browser.browserAction.setIcon({path: `icons/${variation.name}.svg`});
-    browser.browserAction.setTitle({title: variation.name});
-    browser.browserAction.onClicked.addListener(() => this.handleButtonClick());
-  }
-
-  /** handleButtonClick
-    *
-    * - instrument browserAction button clicks
-    * - change label
-    */
-  handleButtonClick() {
-    // note: doesn't persist across a session, unless you use localStorage or similar.
-    this.timesClickedInSession += 1;
-    console.log("got a click", this.timesClickedInSession);
-    browser.browserAction.setBadgeText({text: this.timesClickedInSession.toString()});
-
-    // telemetry: FIRST CLICK
-    if (this.timesClickedInSession == 1) {
-      this.telemetry({"event": "button-first-click-in-session"});
-    }
-
-    // telemetry EVERY CLICK
-    telemetry({"event": "button-click", timesClickedInSession: ""+this.timesClickedInSession});
-
-    // webExtension-initiated ending for "used-often"
-    //
-    // - 3 timesClickedInSession in a session ends the study.
-    // - see `../Config.jsm` for what happens during this ending.
-    if (this.timesClickedInSession >= 3) {
-      msgStudyUtils("endStudy", {reason: "used-often"});
-    }
-  }
-}
-
-/** CONFIGURE and INSTRUMENT the BrowserAction button for a specific variation
- *
- *  1. Request 'info' from the hosting Legacy Extension.
- *  2. We only care about the `variation` key.
- *  3. initialize the feature, using our specific variation
- */
 function runOnce() {
-  msgStudyUtils("info").then(
-    ({variation}) => new BrowserActionButtonChoiceFeature(variation)
-  ).catch(function defaultSetup() {
-    // Errors here imply that this is NOT embedded.
-    console.log("you must be running as part of `web-ext`.  You get 'corn dog'!");
-    new BrowserActionButtonChoiceFeature({"name": "isolatedcorndog"})
-  });
+  // We don't do much here, since we don't really use non-legacy stuff.
+  // We keep this file around to not mess with the template structure.
```

Comment:
I'd recommend removing the embedded web extension if you aren't using, and removing the code in in bootstrap.js. I wouldn't see the Shield study example as a template, but more of a starting point. You don't have to keep everything in it.

Reply:
Great point! I didn't know that was possible at all. I'll take care of this.